agent/src/main/java/com/microsoft/applicationinsights/agent/internal/CommonsLogFactoryClassFileTransformer.java

```diff
@@ -33,7 +33,8 @@
 
 import static org.objectweb.asm.Opcodes.ASM7;
 
-// this is needed because gradle shadow doesn't shade META-INF/services/org.apache.commons.logging.LogFactory
+// this is needed because gradle shadow doesn't shade the constant
+// "META-INF/services/org.apache.commons.logging.LogFactory" that is in commons-logging code
```

Comment:
shadowJar does have a method for shading SPI files. Did it not work in this case?

Reply:
it shaded the SPI file, but it didn't shade the constant that was in the .class file, so the .class file still pointed to `META-INF/services/org.apache.commons.logging.LogFactory`

Reply:
ah, I see. I didn't think there was an explicit reference to this file.